src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,101 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class RegistrationServiceImplTest {
+
+    private RegistrationServiceImpl registrationService;
+
+    @BeforeEach
+    public void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        registrationService.register(null);
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        User user = new User();
+        user.setLogin("abc");
+        user.setPassword("password");
+        user.setAge(20);
+        registrationService.register(user);
+    }
+
+    @Test
+    void register_duplicateLogin_notOk() {
+        User user1 = new User();
+        user1.setLogin("uniqueLogin");
+        user1.setPassword("password");
+        user1.setAge(20);
+        registrationService.register(user1);
+
+        User user2 = new User();
+        user2.setLogin("uniqueLogin");
+        user2.setPassword("password123");
+        user2.setAge(21);
+        registrationService.register(user2);
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("123");
+        user.setAge(20);
+        registrationService.register(user);
+    }
+
+    @Test
+    void register_simplePassword_notOk() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("simple");
+        user.setAge(20);
+        RegistrationException registrationException = Assertions.assertThrows(RegistrationException.class, () -> registrationService.register(user));
+        Assertions.assertEquals("The password must be complex!", registrationException.getMessage());
+    }
+
+    @Test
+    void register_ComplexPassword_Ok() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("password@/123");
+        user.setAge(20);
+        User registeredUser = registrationService.register(user);
+        assertNotNull(registeredUser);
+    }
+
+    @Test
+    void register_underageUser_notOk() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("validPassport");
+        user.setAge(17);
+        registrationService.register(user);
+    }
+
+    @Test
+    void register_validUser_Ok() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("validPassport");
+        user.setAge(20);
+
+        User registeredUser = registrationService.register(user);
+
+        assertNotNull(registeredUser);
+        assertEquals(user.getLogin(), registeredUser.getLogin());
+        assertEquals(user.getPassword(), registeredUser.getPassword());
+        assertEquals(user.getAge(), registeredUser.getAge());
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
```

Comment:
You are missing some test cases. For example, you should add tests for all possible User's parameters (null login/password/age, user under 18/18/over 18 years old, negative age, and so on...). Also, don't forget to check all possible edge cases. For example, we have min password length of 6 characters. In this case would be good to test: empty password (0-length - not-ok), 'abc' (3-length - not-ok), 'abcdf' (5-length - **not-ok edge case**), 'abcdef' (6-length - **ok edge case**), 'abcdefgh' (8-length - ok). Few not-ok checks you can have in one not-ok test method. Same for ok-test.